CBinding/CMake/CMakeProject.cs

```diff
@@ -43,10 +43,53 @@ public class CMakeProject : FolderBasedProject
 		string name;
 		FilePath outputDirectory = new FilePath ("./bin");
 		CMakeFileFormat fileFormat;
+		CMakeToolchain cmakeToolchain;
+
+		/// <summary>
+		/// Occurs when a file is removed from this project.
+		/// </summary>
+		public event ProjectFileEventHandler FileRemovedFromProject;
+
+		/// <summary>
+		/// Occurs when a file is added to this project.
+		/// </summary>
+		public event ProjectFileEventHandler FileAddedToProject;
+
+		/// <summary>
+		/// Occurs when a file of this project has been modified
+		/// </summary>
+		public event ProjectFileEventHandler FileChangedInProject;
+
+		/// <summary>
+		/// Occurs when a file of this project has been renamed
+		/// </summary>
+		public event ProjectFileRenamedEventHandler FileRenamedInProject;
+
+
+		public bool HasLibClang { get; private set; }
+
+		public CLangManager ClangManager { get; private set; }
+
+		public SymbolDatabaseMediator DB { get; private set; }
+
+		public UnsavedFilesManager UnsavedFiles { get; private set; }
+
+		ProjectItemCollection items;
+		/// <summary>
+		/// Files of the project
+		/// </summary>
+		public ProjectFileCollection Files {
+			get { return projectFiles; }
+		}
+		public ProjectFileCollection projectFiles;
+
+		public ProjectItemCollection Items {
+			get { return items; }
+		}
 
 		static readonly string [] supportedLanguages = { "C", "C++", "Objective-C", "Objective-C++" };
 
-		Regex extensions = new Regex (@"(\.c|\.c\+\+|\.cc|\.cpp|\.cxx|\.m|\.mm|\.h|\.hh|\.h\+\+|\.hm|\.hpp|\.hxx|\.in|\.txx)$",
+		public static Regex extensions = new Regex (@"(\.c|\.c\+\+|\.cc|\.cpp|\.cxx|\.m|\.mm|\.h|\.hh|\.h\+\+|\.hm|\.hpp|\.hxx|\.in|\.txx)$",
```

Comment:
Instead of making this public, I would suggest adding a `public bool IsCFile(bool filename)`.

That way, code that uses it will be more readable - `CMakeProject.IsCFile(foo)` instead of `CMakeProject.extensions.IsMatch(foo)`